Source/x11/XGServerWindow.m

```diff
@@ -2602,6 +2605,28 @@ - (void) miniwindow: (int) win
     XIconifyWindow(dpy, window->ident, window->screen);
 }
 
+/* Actually this is "hide application" action.
+   However, key press may be received by particular window. */
+- (BOOL) hidewindow: (int)win
```

Comment:
It doesn't hide application actually. It sends message to window manager on window event. Also, I assume that this is a specific style in XGServerWindow: for example, the method above should be called `miniaturizewindow` instead of `miniwindow` (FYI `miniwindow` in OpenStep called icon window itself). It's quite confusing.

Reply:
I think this method naming came from Display PostScript style of early implementation of GNUstep backend. I think we should rename all methods in one style or keep current. This patch is not about changing methods naming style.

Reply:
You are correct that some of the other method names are not ideal and can only be explained by history. But is that an argument to give a new method a name that doesn't have anything to do with what the method does?
And no, this method isn't just called when a window get hidden. It gets called when the application should be hidden with the main menu as argument.

Reply:
Note that method argument is a window. To be more pedantic, this method should be named like this `-hideApplicationForWindow:(int)windowNumber`. If it's OK, isn't it confusing to have name out of the style of other methods?

Actually x11 part of GNUstep should be refactored. Indentation in many places is a mess and it's hard to read a code. Are you sure it's good time to discuss name of just one new method?

BTW, the comment I've inserted should give a clue about this method to future refactoring.

Reply:
Actually that is quite a nice name. Why not use that? Or if it is too long for you, leave out the window argument, call it -hideApplication and get the window in the backend via NSApp.
With either of this changes the two pull requests would be ready for merge for me.

Reply:
Done.